### PR TITLE
Add functionality to create, look up, edit and delete notes

### PR DESCRIPTION
**Pull request by HelHeim111**

### CODE REVIEW

File: notes_book.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+from note import Note
```
Comment (ib-ua) [inline notes_book.py]: Imported but never used

File: notes_book.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+class Note_book:
```
Comment (ib-ua) [inline notes_book.py]: class name doesn't match code convention

File: notes_book.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    def __init__(self):
+        self._notes = {}
+    
+    @property
+    def notes(self):
+        return self._notes
+    
+    @notes.setter
```
Comment (ib-ua) [inline notes_book.py:9]: names of the function and decorator don't match

File: notes_book.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    def set_notes(self, notes):
+        if notes is not None and type(notes) is dict:
+            self._notes = notes
```
Comment (ib-ua) [inline notes_book.py]: prefix private fields with double _ character

File: note.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
+class Note:
```
Comment (ib-ua) [inline note.py:1]: The class Note migth be descendants from Field

File: notes_book.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    
+    def find_notes_author(self, author):
+        return self._notes.get(author, [])
+        
+    def find_notes_author_title(self, author, title):
+        notes_by_author = self.find_notes_author(author)
+        if notes_by_author is not None:
+            return [note for note in notes_by_author if note.title.lower() == title.lower()]
+        else:
+            return []
```
Comment (ib-ua) [inline notes_book.py]: It doesn't make sense to return an empty list in case of reading notes for an unknown user

**Review by ib-ua — CHANGES_REQUESTED** [review]

**Review by ib-ua — APPROVED** [review]

**Review by Flairim — APPROVED** [review]

**Pull request merged**